chef/cookbooks/nova/recipes/compute.rb

```diff
@@ -130,19 +130,41 @@ def set_boot_kernel_and_trigger_reboot(flavor='default')
   end
 
   case node[:nova][:libvirt_type]
-    when "kvm"
-      package "kvm"
+    when "kvm", "qemu"
+      if node.platform_version.to_f >= 12.0
+        package "qemu"
+        package "qemu-kvm"
+        package "qemu-block-rbd"
+      else
+        package "kvm"
+      end
+
+      # Install Ceph integration if needed
+      cinder_servers = search(:node, "roles:cinder-controller") || []
```

Comment:
well, the config.rb code can't be moved easily here as it uses the code for determining the secret uuid (whcih needs to be written to nova.conf)

I don't see an easy way to avoid the code duplication, and then again its just a two lines duplication which I find acceptable. 

it will eventually go away once the "ephemeral disks on ceph" feature is implemented. for now it would be a good idea to have the code there to auto setup the nodes though.